irohad/consensus/yac/cluster_order.hpp

```diff
@@ -39,7 +39,7 @@ namespace iroha {
         /**
          * Provide current leader peer
          */
-        const shared_model::interface::Peer &currentLeader();
+        const std::shared_ptr<shared_model::interface::Peer> &currentLeader();
```

Comment:
I would make
```suggestion
        std::shared_ptr<shared_model::interface::Peer> currentLeader();
```
as returning a reference to a local object is scary, and then
```suggestion
        std::shared_ptr<const shared_model::interface::Peer> currentLeader();
```
as we do not want anyone to change the object (no matter it is an interface)